networkx/algorithms/chordal.py

```diff
@@ -411,33 +413,68 @@ def complete_to_chordal_graph(G):
     >>> G = nx.wheel_graph(10)
     >>> H, alpha = complete_to_chordal_graph(G)
     """
+    if is_chordal(G):
+        H = G.copy()
+        alpha = dict.fromkeys(H, 0)
+    else:
+        H, alpha = _triangulate_fully(G)
+
+    return H, alpha
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+@nx._dispatchable
+def minimal_elimination_order(G):
+    # TODO: Add docstring
+    return _triangulate_fully(G)
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+@nx._dispatchable
+def perfect_elimination_order(G):
+    # TODO: Add docstring
+    if not is_chordal(G):
+        raise nx.NetworkXError("Input graph is not chordal.")
+
+    return _triangulate_fully(G)
+
+
+def _triangulate_fully(G):
+    # TODO: Add docstring
     H = G.copy()
     alpha = dict.fromkeys(H, 0)
-    if nx.is_chordal(H):
-        return H, alpha
     chords = set()
-    weight = dict.fromkeys(H.nodes(), 0)
-    unnumbered_nodes = list(H.nodes())
-    for i in range(len(H.nodes()), 0, -1):
+
+    nodes = H.nodes()
+    weights = dict.fromkeys(nodes, 0)
+    unnumbered_nodes = list(nodes)
+    n = len(unnumbered_nodes)
```

Comment:
No need to create multiple objects to hold the nodes.
```suggestion
    weights = dict.fromkeys(H, 0)
    unnumbered_nodes = list(H)
    n = len(unnumbered_nodes)
```